src/StoreApi/Routes/Checkout.php

```diff
@@ -375,6 +374,23 @@ private function create_or_update_draft_order() {
 			$order_controller->update_order_from_cart( $this->order );
 		}
 
+		/**
+		 * WooCommerce Blocks Checkout Update Order Meta (experimental).
+		 *
+		 * This hook gives extensions the chance to add or update meta data on the $order.
+		 *
+		 * This is similar to existing core hook woocommerce_checkout_update_order_meta.
+		 * We're using a new action:
+		 * - To keep the interface focused (only pass $order, not passing request data).
+		 * - This also explicitly indicates these orders are from checkout block/StoreAPI.
+		 *
+		 * @see https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3686
+		 * @internal This Hook is experimental and may change or be removed.
+		 *
+		 * @param WC_Order $order Order object.
+		 */
+		do_action( '__experimental_woocommerce_blocks_checkout_update_order_meta', $this->order );
+
```

Comment:
We did a lot of testing yesterday in our call so I'm confident that this works, I can't comment yet on the payment change until I understand it a bit better but approving this part.